src/middlewared/middlewared/plugins/disk.py

```diff
@@ -1061,6 +1101,61 @@ def label_to_dev(self, label, geom_scan=True):
 
         await self.sync(dev)
 
+    @private
+    def format(self, disk, swapgb):
+
+        job = self.middleware.call_sync('disk.wipe', disk, 'QUICK')
```

Comment:
How about `middleware.call_job`/`middleware.call_job_sync` that will wait for job and raise exception?

Reply:
Not sure I am fan of that idea, perhaps have the job.wait* do that by default?

Either way, we do that kind of stuff in other places as well, perhaps can be tackled in another PR/day?